src/shared/store/actions/utils/errors.ts

```diff
@@ -0,0 +1,33 @@
+export class Forbidden extends Error {
+  readonly code: number = 403;
+}
+export class NotFound extends Error {
+  readonly code: number = 404;
+}
+export class ServiceUnavailable extends Error {
+  readonly code: number = 503;
+}
+export class BadRequest extends Error {
+  readonly code: number = 400;
+}
+
+export type RequestErrors =
+  | BadRequest
+  | Forbidden
+  | NotFound
+  | ServiceUnavailable;
+
+export const requestErrorTypes: { [errorName: string]: any } = {
+  BadRequest,
+  Forbidden,
+  NotFound,
+  ServiceUnavailable,
+};
+
+export const formatError = (error: Error) => {
+  const errorType = error.message.replace(/ /g, '');
+  if (requestErrorTypes[errorType]) {
+    return new requestErrorTypes[errorType](error.message);
+  }
+  return new BadRequest(error.message);
```

Comment:
This would return an erroneous code if the individual status code is not explicitely handled in the list below. This problem can be solved if using my proposal of making RequestError a class and individual error types, well, types